oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileStoreIterator.java

```diff
@@ -121,8 +121,8 @@ protected NodeStateEntry computeNext() {
     private NodeStateEntry computeNextEntry() {
         if (!buffer.isEmpty()) {
             if (buffer.size() > maxBufferSize || buffer.estimatedMemoryUsage() > maxBufferSizeBytes) {
-                maxBufferSize = buffer.size();
-                maxBufferSizeBytes = buffer.estimatedMemoryUsage();
+                maxBufferSize = Math.max(buffer.size(), maxBufferSize);
```

Comment:
Im not against this change, but I'm wondering if it strictly necessary? Seems like the `bufferSize` will be larger than `maxBufferSize` if we enter this condition either way.

Reply:
We enter this condition if either `buffer.size() > maxBufferSize` or `buffer.estimatedMemoryUsage() > maxBufferSizeBytes`. And these two conditions are independent. For instance, we can have the following sequence:

`buffer.size() = 10, buffer.estimatedMemoryUsage() = 1000`
10 elements of size 1000 in total.

And then 
`buffer.size() = 1, buffer.estimatedMemoryUsage() = 2000`
1 element of size 2000.

So in this case, we want to update the max memory estimate but leave the max size at 10.

Reply:
Yes the change makes sense.

We could also have to "if" statements and two updates methods (without max), but then we would need two log statements. Your change is a bit shorter (even thought, harder to understand).

But the old behavior looks incorrect.